File: QB_Challenges/PrepInsta/JAVA/01_Negative_Or_Positive/NOP_Ternary.java
import java.util.Scanner;

public class NOP_Ternary {
    
    public static void NOP() {

        Scanner sc = new Scanner(System.in);
        System.out.print("Enter your number: ");
        int num = sc.nextInt();
        sc.close();
        
        String result =
        (num > 0) ? "The given number "+ num + " is a postive number.":
        (num < 0) ? "The given number "+ num + " is a negative number":
        "The given number "+ num +" is zero";
        
        System.out.println(result);
                    
        
    }

    public static void main(String[] args) {
        NOP();
    }
}

File: QB_Challenges/PrepInsta/CPP/01_Negative_Or_Positive/NOP_Ternary.cpp
#include <iostream>
using namespace std;

void NOP(){
    int num;
    cout <<"Enter the number: ";
    cin >>num;

    string result = 
    (num > 0) ? to_string(num) + " is a positive number." :
    (num < 0) ? to_string(num) + " is a negative number." :
    to_string(num) + " is Zero number";

    cout <<result <<endl;

}

int main(){
NOP();
  return 0;
}